account-kit/smart-contracts/src/ma-v2/utils.ts

```diff
@@ -261,7 +261,7 @@ export const buildFullNonceKey = ({
 }: BuildNonceParams): bigint => {
   return (
     (nonceKey << 40n) +
-    BigInt(entityId << 8) +
+    (BigInt(entityId) << 8n) +
```

Comment:
fixes an overflow bug when trying to shift the `entityId` before converting it to `BigInt`